server/properties.lua

```diff
@@ -1 +1,541 @@
+end
+
+-- Enter furnished property
+RegisterNetEvent('qbx-property:server:enterProperty', function(propertyId, isVisit)
+    local source = source
+    local playersToConceal = {}
+    local playersInsideProperty = {}
+    local property = properties[propertyId]
+    if property.property_type ~= 'ipl' then return end -- remove when shells are implemented
+    for k, v in pairs(properties) do
+        if v.property_type == 'ipl' then
+            if k == propertyId then
+                for _, serverid in pairs(v.playersInside) do
+                    playersInsideProperty[#playersInsideProperty + 1] = serverid
+                end
+            else
+                for _, serverid in pairs(v.playersInside) do
+                    playersToConceal[#playersToConceal + 1] = serverid
+                end
+            end
+        end
+    end
+
+    if property.property_type == 'ipl' then
+        TriggerClientEvent('qbx-property:client:enterIplProperty', source, property.interior, propertyId, isVisit, property.options or false)
+    else
+        return -- do nothing for now
+        -- TODO: shell stuff (have fun with that)
+    end
+    TriggerClientEvent('qbx-property:client:concealPlayers', source, playersToConceal, true)
+    TriggerClientEvent('qbx-property:client:concealPlayers', -1, {source}, true)
+    for _, v in pairs(playersInsideProperty) do
+        TriggerClientEvent('qbx-property:client:concealPlayers', v, {source}, false)
+    end
+    Player(source).state:set('inProperty', {propertyid = propertyId}, true)
+end)
+
+-- Enter garage
+---WIP
+RegisterNetEvent('qbx-property:server:enterGarage', function(propertyId, isVisit, isInVehicle)
+    if not propertyId then return end
+    local property = properties[propertyId]
+    if isVisit then
+        if isInVehicle then
+            return TriggerClientEvent('QBCore:Notify', source, Lang:t('error.inVehicle'), 'error')
+        end
+        TriggerClientEvent('qbx-property:client:enterIplProperty', source, property.interior, propertyId, true, property.options or false)
+        return
+    end
+
+    local playersToConceal = {}
+    local playersInsideProperty = {}
+    for k, v in pairs(properties) do
+        if v.property_type == 'garage' then
+            if k == propertyId then
+                for _, serverid in pairs(v.playersInside) do
+                    playersInsideProperty[#playersInsideProperty + 1] = serverid
+                end
+            else
+                for _, serverid in pairs(v.playersInside) do
+                    playersToConceal[#playersToConceal + 1] = serverid
+                end
+            end
+        end
+    end
+
+    if isInVehicle then
+        return -- Do nothing for now
+        -- get the vehicle, add it to the garage and remove it from the world
+        -- add it to the first available slot
+    end
+
+    TriggerClientEvent('qbx-property:client:concealPlayers', source, playersToConceal, true)
```

Comment:
Does this always need to happen when entering a garage? If so, why not handle in the event handler for garage entry?

Reply:
yes, but is it worth it to add another argument to the handler when you can just use a separate handler?
Technically it's just extracting events i guess?

Reply:
One extra argument isn't a burden at all

Reply:
> One extra argument isn't a burden at all

I guess it's more that if it's easy to extract an event why not do it